.github/workflows/test-check.yml

```diff
@@ -0,0 +1,78 @@
+name: Test Checker
+
+on:
+  pull_request:
+    branches:
+      - master
+      - release/*
+
+jobs:
+  test_check:
+    name: Check test results
+    runs-on: ubuntu-latest
+    steps:
+      - name: Wait for build to succeed ( ubuntu )
+        uses: tomchv/wait-my-workflow@v1.1.0
+        id: wait-for-build-ubu
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          checkName: "Tests & Build (ubuntu-latest)"
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
+          timeoutSeconds: 7200
+
+      - name: Make sure the build was a success ( ubuntu )
+        if: steps.wait-for-build-ubu.outputs.conclusion != 'success'
+        run: echo buils not success && false
+
+      - name: Wait for build to succeed ( windows )
+        uses: tomchv/wait-my-workflow@v1.1.0
+        id: wait-for-build-win
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          checkName: "Tests & Build (windows-latest)"
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
+          timeoutSeconds: 7200
+
+      - name: Make sure the build was a success ( windows )
+        if: steps.wait-for-build-win.outputs.conclusion != 'success'
+
+        run: echo buils not success && false
+      - name: Wait for build to succeed ( macos )
+        uses: tomchv/wait-my-workflow@v1.1.0
+        id: wait-for-build-mac
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          checkName: "Tests & Build (macos-latest)"
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
+          timeoutSeconds: 7200
+
+      - name: Make sure the build was a success ( macos )
```

Comment:
nitpick: macos is usually faster than windows, so we can swap order

Reply:
Somewhat related: @maaktweluit is the order here relevant? In other words: if a build is finished before we get to this step will the result be reported correctly?

Reply:
1. fixed
2. the order does not matter at all, if the build already succeeded ( for example on rebuild ) it returns green right away.